test/functional/wallet_upgradetohd.py

```diff
@@ -62,7 +62,7 @@ def run_test(self):
                     assert_equal(keypath, "m/44'/1'/0'/0/%d" % i)
                 else:
                     keypath = node.getaddressinfo(out['scriptPubKey']['addresses'][0])['hdkeypath']
-                    assert_equal(keypath, "m/44'/1'/0'/1/%d" % i)
+                    assert_equal(keypath, "m/44'/1'/0'/1/%d" % (i * 2))
```

Comment:
14582: why are there changes in `test/functional/wallet_upgradetohd.py`?

Reply:
Due to 2 tries to create a transaction 2 private keys are used now. See release note and changes in `src/wallet/wallet.cpp`

Reply:
I will try to backport with higher priority this one https://github.com/bitcoin/bitcoin/pull/27053 which resolve this issue but is implemented 4 major version of bitcoin later (v25) and causes bunch of conflicts.